lib/rules/block-spacing.js

```diff
@@ -15,7 +15,8 @@ var util = require("../ast-utils");
 module.exports = function(context) {
     var always = (context.options[0] !== "never"),
         message = always ? "Requires a space" : "Unexpected space(s)",
-        sourceCode = context.getSourceCode();
+        sourceCode = context.getSourceCode(),
+        tokensAndComments = sourceCode.tokensAndComments;
```

Comment:
Yes, SourceCode is part of the public API.  I don't see any reason to deprecate this, we just need a better way to get at individual tokens/comments than indexOf.